### PR TITLE
Expose OS-chosen address and port; force IPv4

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
+project(CPP-NETLIB VERSION 0.11.1)
 
-option( CPP-NETLIB_BUILD_SHARED_LIBS "Build cpp-netlib as shared libraries." OFF )
-option( CPP-NETLIB_BUILD_TESTS "Build the cpp-netlib project tests." ON)
-option( CPP-NETLIB_BUILD_EXPERIMENTS "Build the cpp-netlib project experiments." ON)
-option( CPP-NETLIB_BUILD_EXAMPLES "Build the cpp-netlib project examples." ON)
+option( CPP-NETLIB_BUILD_SHARED_LIBS "Build cpp-netlib as shared libraries." ON )
+option( CPP-NETLIB_BUILD_TESTS "Build the cpp-netlib project tests." OFF)
+option( CPP-NETLIB_BUILD_EXPERIMENTS "Build the cpp-netlib project experiments." OFF)
+option( CPP-NETLIB_BUILD_EXAMPLES "Build the cpp-netlib project examples." OFF)
```
Comment (deanberris) [inline CMakeLists.txt]: I'm not comfortable with changing these defaults. Is there a reason why you don't set these from your project?

File: boost/network/protocol/http/server/sync_server.hpp [hunk 2]
```diff
@@ -86,7 +89,7 @@ struct sync_server_base : server_storage_base, socket_options_base {
     using boost::asio::ip::tcp;
     system::error_code error;
     tcp::resolver resolver(service_);
-    tcp::resolver::query query(address_, port_);
+    tcp::resolver::query query(tcp::v4(), address_, port_);
```
Comment (deanberris) [inline boost/network/protocol/http/server/sync_server.hpp]: There seems to be no reason for this -- can you not make this explicit in your code instead? As part of the options?

**Comment by deanberris** [comment]
Hi @eile -- I sent some comments a while back and I'm unclear as to why this is necessary and why this isn't an optional thing (as opposed to something that forces IPv4). Do you mind making it so that we can configure what kind of IP addresses we'd like to get from the DNS would be? Then I can go ahead and merge the updated version.

**Comment by eile** [comment]
Hi @deanberris: @tribal-tec and me where a bit swamped lately, we'll get back to you on this.

**Comment by deanberris** [comment]
All good, thanks @eile -- no rush. :)

**Comment by deanberris** [comment]
Also by the way, I just noticed that you're attempting to make a change to the 0.11-devel branch, which we aren't actively maintaining anymore. We're trying to get everyone to use the 0.13-release branch instead. I'm happy to still merge an updated version of this in 0.11-devel, but I'm not really up for having another release on the 0.11.x line.

**Comment by tribal-tec** [comment]
Please close this in favor of https://github.com/cpp-netlib/cpp-netlib/pull/729

**Pull request closed without merging**